File: hard/14_non_attacking_queens/non_attacking_queens_test.go
package main

import "testing"

func TestProgram(t *testing.T) {
	input := 4
	expected := 2
	actual := nonAttackingQueens(input)
	if actual != expected {
		t.Errorf("nonAttackingQueens(%d) = %d; expected %d", input, actual, expected)
	}
}

File: hard/14_non_attacking_queens/non_attacking_queens.test.js
const nonAttackingQueens =  require('./non_attacking_queens.js');

describe('nonAttackingQueens', () => {
  it('should return the number of non attacking queens', () => {
    const input = 4;
    const expected = 2;
    const actual = nonAttackingQueens(input);
    expect(actual).toEqual(expected);
  });
});